core/src/main/scala/kafka/log/Log.scala

```diff
@@ -756,10 +761,10 @@ class Log(@volatile var dir: File,
    *
    * @return The newly rolled segment
    */
-  def roll(): LogSegment = {
+  def roll(offset: Long): LogSegment = {
```

Comment:
Could we make offset defaults to 0 here? Also, would it be better to name offset sth like expectedNextOffset?